packages/btc/src/api/sendRbf.ts

```diff
@@ -0,0 +1,169 @@
+  builder: TxBuilder;
+  fee: number;
+  feeRate: number;
+  changeIndex: number;
+}> {
+  const previousTx = bitcoin.Transaction.fromHex(props.txHex);
+  const network = networkTypeToNetwork(props.source.networkType);
+
+  // Rebuild inputs
+  const inputs: Utxo[] = [];
+  for (const input of previousTx.ins) {
+    const hash = Buffer.from(input.hash).reverse().toString('hex');
+    const utxo = await props.source.getUtxo(hash, input.index);
+    if (!utxo) {
+      throw TxBuildError.withComment(ErrorCodes.CANNOT_FIND_UTXO, `hash: ${hash}, index: ${input.index}`);
+    }
+
+    // Ensure each P2TR input has a corresponding pubkey
+    const fromPubkey = utxo.address === props.from ? props.fromPubkey : undefined;
+    const inputPubkey = props.inputsPubkey?.[utxo.address];
+    const pubkey = inputPubkey ?? fromPubkey;
+    if (pubkey) {
+      utxo.pubkey = pubkey;
+    }
+    if (isP2trScript(utxo.scriptPk) && !utxo.pubkey) {
+      throw TxBuildError.withComment(ErrorCodes.MISSING_PUBKEY, utxo.address);
+    }
+
+    inputs.push(utxo);
+  }
+
+  // Rebuild outputs
+  const requireValidOutputsValue = props.requireValidOutputsValue ?? false;
+  const outputs: InitOutput[] = previousTx.outs.map((output) => {
+    if (isOpReturnScriptPubkey(output.script)) {
+      return {
+        script: output.script,
+        value: output.value,
+        fixed: true,
+      };
+    } else {
+      return {
+        minUtxoSatoshi: requireValidOutputsValue ? undefined : output.value,
+        address: bitcoin.address.fromOutputScript(output.script, network),
+        value: output.value,
+        fixed: true,
+      };
+    }
+  });
+
+  // Set change output if specified
+  let changeAddress: string | undefined = props.changeAddress;
+  if (props.changeIndex !== undefined) {
+    const changeOutput = outputs[props.changeIndex];
+    if (!changeOutput) {
+      throw TxBuildError.withComment(ErrorCodes.INVALID_CHANGE_OUTPUT, `outputs[${props.changeIndex}] is not found`);
+    }
+    const isReturnableOutput = changeOutput && 'address' in changeOutput;
+    if (!isReturnableOutput) {
+      throw TxBuildError.withComment(
+        ErrorCodes.INVALID_CHANGE_OUTPUT,
+        `outputs[${props.changeIndex}] is not a returnable output for change`,
+      );
+    }
+    const changeOutputAddress = isReturnableOutput ? changeOutput.address : undefined;
+    if (changeOutputAddress && changeAddress && changeAddress !== changeOutputAddress) {
+      throw TxBuildError.withComment(
+        ErrorCodes.INVALID_CHANGE_OUTPUT,
+        `outputs[${props.changeIndex}].address does not match the specified changeAddress`,
+      );
+    }
+    if (changeOutputAddress && !changeAddress) {
+      changeAddress = changeOutputAddress;
+    }
+
+    const config = networkTypeToConfig(props.source.networkType);
+    const minUtxoSatoshi = props.minUtxoSatoshi ?? config.btcUtxoDustLimit;
+    changeOutput.minUtxoSatoshi = minUtxoSatoshi;
+    changeOutput.value = minUtxoSatoshi;
```

Comment:
### Discussion
If the changeIndex is the last index of previousTx.outs, do you think it is safe to remove this change output?

Reply:
I was dedicated to keeping the structure of the RBF transaction as close to the original transaction as possible by ensuring that each input and output from the original transaction exists in the RBF transaction. However, logically speaking, removing the last output (if specified by the `changeIndex` option) sounds like a workable idea, I'll test it.

The logical difference:
- The current logic: the RBF_TX will retain the change output if it already exists
- Remove the last output: if `fee == inputs - outputs`, the RBF_TX will not have a change output at all

Reply:
Updated logic and tests at: [d886b25](https://github.com/ckb-cell/rgbpp-sdk/pull/150/commits/d886b258dcd8406cf4e00deb71c4be96b133c871), [b8a99b6](https://github.com/ckb-cell/rgbpp-sdk/pull/150/commits/b8a99b6324aa7c0df899db7078308a56a0fea7d5)